v2/controllers/marketplace/clusterregistration_controller.go

```diff
@@ -435,9 +450,63 @@ func (r *ClusterRegistrationReconciler) SetupWithManager(mgr ctrl.Manager) error
 					if _, ok := secret.Data[utils.IBMEntitlementDataKey]; ok && secretName == utils.IBMEntitlementKeySecretName {
 						return true
 					}
+
+					if secretName == utils.RHM_OPERATOR_SECRET_NAME {
+						return true
+					}
+
 					return false
 				},
 			},
 		)).
+		// This controller is responsible for keeping the AccountID & Env annotation updated on MarketplaceConfig (connected)
+		// Since the values are dervived from the pull secret
+		// It also must reconcile due a change from IsDisconnected to connected mode
+		Watches(
+			&source.Kind{Type: &marketplacev1alpha1.MarketplaceConfig{}},
+			handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
+				return []reconcile.Request{
+					{NamespacedName: types.NamespacedName{
+						Name:      utils.RHMPullSecretName,
+						Namespace: a.GetNamespace(),
+					}},
+				}
+			}),
+			builder.WithPredicates(predicate.Funcs{
+				// Queue the reconciler in a connected environment
+				// Handle the case where start in disconnected mode, and switch to connected
+				CreateFunc: func(e event.CreateEvent) bool {
+					marketplaceConfig, ok := e.Object.(*marketplacev1alpha1.MarketplaceConfig)
+					if ok {
+						return !*marketplaceConfig.Spec.IsDisconnected
+					}
+					return false
+				},
+				UpdateFunc: func(e event.UpdateEvent) bool {
+
+					marketplaceConfigNew, newOk := e.ObjectNew.(*marketplacev1alpha1.MarketplaceConfig)
+					marketplaceConfigOld, oldOk := e.ObjectOld.(*marketplacev1alpha1.MarketplaceConfig)
+
+					if newOk && oldOk {
+						if *marketplaceConfigNew.Spec.IsDisconnected == false && *marketplaceConfigOld.Spec.IsDisconnected == true {
+							return true
+						}
+
+						if !*marketplaceConfigNew.Spec.IsDisconnected {
+							if marketplaceConfigNew.Spec.RhmAccountID != marketplaceConfigOld.Spec.RhmAccountID {
+								return true
+							}
```

Comment:
if we are in a connected env, and the user has modified the rhmaccountid, we should trigger this reconciler to resolve it back to the correct accountid which we can get because we have the pull secret